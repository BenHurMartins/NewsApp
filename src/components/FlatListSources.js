import React, {useEffect, useState} from 'react';
import {FlatList, StyleSheet, Text} from 'react-native';
import {Mixins} from '../styles';
import {getSources} from '../services/sources/sources';
import {SourceItemComponent} from './';
import {useDeviceOrientation} from '@react-native-community/hooks';

const FlatListSources = () => {
  const [sources, setSources] = useState([]);
  const renderItem = ({item}) => <SourceItemComponent source={item} />;

  const orientation = useDeviceOrientation();
  const dinamicWidth = orientation.portrait
    ? Mixins.DEVICE_WIDTH
    : Mixins.DEVICE_HEIGHT / 2 - 60;

  useEffect(async () => {
    await getSources().then(response => {
      setSources(response.data.sources);
    });
  }, []);

  return (
    <FlatList
      contentContainerStyle={
        ([styles.flatListContainer], {width: dinamicWidth})
      }
      data={sources}
      renderItem={renderItem}
      keyExtractor={(item, index) => item.id}
      // onEndReached={getMoreResults}
      onEndReachedThreshold={0.7}
    />
  );
};

export default FlatListSources;

const styles = StyleSheet.create({
  flatListContainer: {
    alignItems: 'center',
  },
});
