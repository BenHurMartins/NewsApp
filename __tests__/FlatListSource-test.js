import 'react-native';
import React from 'react';
import FlatListSources from '../src/components/FlatListSources';
import renderer from 'react-test-renderer';

describe('FlatListSources component', () => {
  it('FlatListSources renders correctly', () => {
    renderer.create(<FlatListSources />);
  });
  it('FlatListSources snapshot test', () => {
    const tree = renderer.create(<FlatListSources />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
