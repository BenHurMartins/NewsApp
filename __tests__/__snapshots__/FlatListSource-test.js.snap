// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlatListSources component FlatListSources snapshot test 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "width": 637,
    }
  }
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  viewabilityConfigCallbackPairs={Array []}
>
  <View />
</RCTScrollView>
`;
